code/pages/DMSDocumentCartCheckoutPage.php

```diff
@@ -0,0 +1,258 @@
+     * fields for the recipients information.
+     *
+     * To extend use the following from within an Extension subclass:
+     *
+     * <code>
+     * public function updateDMSDocumentRequestForm($form)
+     * {
+     *     // Do something here
+     * }
+     * </code>
+     *
+     * @return DMSDocumentRequestForm
+     */
+    public function DMSDocumentRequestForm()
+    {
+        $fields = DMSDocumentCartSubmission::create()->scaffoldFormFields();
+        $fields->replaceField('DeliveryAddressLine2', TextField::create('DeliveryAddressLine2', ''));
+        $fields->replaceField(
+            'DeliveryAddressCountry',
+            CountryDropdownField::create(
+                'DeliveryAddressCountry',
+                _t('DMSDocumentCartCheckoutPage.RECEIVER_COUNTRY', 'Country')
+            )->setValue('NZ')
+        );
+
+        $form = DMSDocumentRequestForm::create($this, 'DMSDocumentRequestForm', $fields, FieldList::create());
+        if ($receiverInfo = $this->getCart()->getReceiverInfo()) {
+            $form->loadDataFrom($receiverInfo);
+        }
+
+        if ($message = $this->request->getVar('message')) {
+            $form->sessionMessage($this->ThanksMessage, 'good');
+        } else {
+            $form->clearMessage();
+        }
+        $this->extend('updateDMSDocumentRequestForm', $form);
+
+        return $form;
+    }
+
+    /**
+     * Sends an email to both the configured recipient as well as the requester. The
+     * configured recipient is bcc'ed to the email in order to fulfill it.
+     *
+     * To extend use the following from within an Extension subclass:
+     *
+     * <code>
+     * public function updateSend($email)
+     * {
+     *     // Do something here
+     * }
+     * </code>
+     * @return mixed
+     *
+     * @throws DMSDocumentCartException
+     */
+    public function send()
+    {
+        $member = $this->CartEmailRecipient();
+        if ($member->exists()) {
+            $cart = $this->getCart();
+            $from = Config::inst()->get('Email', 'admin_email');
+            $emailAddress = ($info = $cart->getReceiverInfo()) ? $info['ReceiverEmail'] : $from;
+            $email = Email::create(
+                $from,
+                $emailAddress,
+                _t('DMSDocumentCartCheckoutPage.EMAIL_SUBJECT', 'Request for Printed Publications')
+            );
+            $email->setBcc($member->Email);
+            $renderedCart = $cart->renderWith('DocumentCart_email');
+            $body = _t(
+                'DMSDocumentCartCheckoutPage.EMAIL_BODY',
+                '<p>A request for printed publications has been submitted with the following details:</p>'
+            );
+            $body .= $renderedCart->getValue();
+            $email->setBody($body)->setReplyTo($emailAddress);
+            $this->extend('updateSend', $email);
+            return $email->send();
+        } else {
```

Comment:
I think it reads better with the else in place. Throwing an exception on a line on its own seems a bit weird.